src/FormattedText.tsx

```diff
@@ -0,0 +1,246 @@
+import * as React from 'react'
+import * as Marked from 'marked'
+import * as He from 'he'
+
+
+interface Props {
+    text: string,
+    format: string
+}
+
+interface State {
+}
+
+export class FormattedText extends React.Component<Props, State> {
+
+    shouldComponentUpdate(nextProps: Props, nextState: State, nextContext: any): boolean {
+        return this.props.text !== nextProps.text || this.props.format !== nextProps.format
+    }
+
+    render(): React.ReactElement<any> {
+        const format = this.props.format || "markdown"
+        if (format === "plain") {
+            return this.renderPlainText()
+        } else if (format === "xml") {
+            return this.renderXml()
+        } else {
+            return this.renderMarkdown()
+        }
+    }
+
+    renderPlainText(): React.ReactElement<any> {
+        // TODO @eanders-MS: This is placeholder until updated to DirectLine 3.0
+        const src = this.props.text || ''
+        const lines = src.replace('\r', '').split('\n')
+        const elements = lines.map(line => <span>{line}<br /></span>)
+        return <span className="format-plain">{elements}</span>
+    }
+
+    renderXml(): React.ReactElement<any> {
+        // TODO @eanders-MS: Implement once updated to DirectLine 3.0
+        return <span className="format-xml"></span>
+    }
+
+    renderMarkdown(): React.ReactElement<any> {
+        let src = this.props.text || ''
+        src = src.replace(/<br\s*\/?>/ig, '\r\n\r\n')
+        const options = {
+            gfm: true,
+            tables: true,
+            breaks: false,
+            pedantic: false,
+            sanitize: true,
+            smartLists: true,
+            silent: false,
+            smartypants: true
+        } as MarkedOptions
+        const renderer = options.renderer = new ReactRenderer(options)
+        const text = Marked(src, options)
+        const elements = renderer.getElements(text)
+        /*// debug
+        const remaining = renderer.elements.filter(el => !!el)
+        if (remaining.length) {
+            console.warn(`There were ${remaining.length} unused markdown elements!`)
+        }*/
+        return <span className="format-markdown">{elements}</span>
+    }
+}
+
+
+class ReactRenderer implements MarkedRenderer {
+
+    elements: React.ReactElement<any>[] = []
+
+    constructor(private options) {
+    }
+
+    /**
+     * We're being sneaky here. Marked is expecting us to render html to text and return that.
```

Comment:
I don't really understand any of this, but I vote we just merge it in as feature work in progress and then discuss next week when I get back.